### ISSUE TITLE
Name missing from posts

### ISSUE THREAD

**Issue by divineaverage** [divineaverage/P7_Groupomania#5]
For this, I am perplexed. I went through the history of Card.jsx and saw that i changed a few things. I changed them back and name still isn't working. I think I'm overlooking something from staring at it too long - do you see anything?

**Comment by AccessiT3ch** [comment]
@divineaverage it looks like you're trying to get the name of the user off the post, instead of the user associated with the post via `authorId`. I'm guessing something like this should work:

```js  
const { authorId, caption, date, imageUrl } = post || {};
const { name } = getUserById(aurthorId);

```

Note: if the user associated with the post isn't already in the store, you'll have to fetch them from the API to add them to the store

**Comment by AccessiT3ch** [comment]
@divineaverage this is looking pretty close! Just be sure to be updating the `lastLogin` property (instead of `date`) and think that should be good if it updates in the DB!

**Comment by divineaverage** [comment]
Okay, I sort of see what's happening - for some reason, profile is not pulling in correctly anymore (it was at some point, as the name was working just fine). It shows as undefined when console logged. I don't think I changed anything there, so I'm not sure why it broke?

**Comment by divineaverage** [comment]
LOL ok - this is weird but may be helpful. One post has the name working. But when I add a new post, the post with the name working changes! It went from one of the cat pics to the happy hour pic to the flower pic lol.... I feel like I'm taking crazy pills....

**Comment by AccessiT3ch** [comment]
@divineaverage - your getUser controller seems to be doing something a little funky: you've got an array (`user` which is returned by `User.find()`) spreading onto an object. It looks like you should be spreading `user[0]` in your controller so you don't have to deal with any arrays in the front-end when adding profiles to state/store.

Not sure if that's causing anything, not yet sure why you'd be getting the behavior you mentioned.

Oh, unless it's the fact that [you're not giving your Card component the key property](https://github.com/divineaverage/P7_Groupomania/blob/main/client/src/components/feed/PostsList.jsx#L47-L53), which is needed when mapping components like you're doing in your feed component. You should be getting an error about this in your console, fixing that may resolve some bizarre behavior (which is why we always give mapped components unique keys).

```js
somethings.map(something => <Something {...something} /> ); // bad

somethings.map(something => <Something key={'asdf-' + something.id} {...something} /> ); // good
```
Forgetting key props can be fine sometimes, other times they present as crazy pills for sure!

**Comment by divineaverage** [comment]
Ok! I got the error to disappear, but now no name shows for any post lol

**Comment by AccessiT3ch** [comment]
@divineaverage i would start console logging the profile object before dispatching it to the store to make sure it looks like what we're expecting. Are the author profiles being added to the redux store correctly?

**Comment by divineaverage** [comment]
Ok, error is back and the profile is undefined. This is all so weird because it was working for awhile without any issues and nothing changed. So I think it's set up right in Redux?

**Comment by AccessiT3ch** [comment]
@divineaverage have you confirmed by looking in the redux developer tool?

**Comment by divineaverage** [comment]
Duh, now I have - it looks like 2 user profiles are being pulled in correctly, then there's a section for undefined. Within that, there's $-- and _doc, and _doc has the most recent post's user info in a different format. However, none of these are pulling the name in correctly on the posts.
![Screenshot 2023-07-23 at 9 30 05 PM](https://github.com/divineaverage/P7_Groupomania/assets/107645919/e2b8fbb6-1670-408f-b148-33d370e35cec)

**Comment by AccessiT3ch** [comment]
@divineaverage ok so probably from your controller you want to send the `profile[0]._doc` back to the front end instead of just `profile[0]`

**Comment by AccessiT3ch** [comment]
@divineaverage what does console logging the profile after getting it from the store in the Card component return?

**Comment by AccessiT3ch** [comment]
Make sure your backend is still running if it crashed? You gotta figure out what is being returned by that User.find, and send the relevant info to the front end in a way that's conducive with the redux slice you set up.

**Comment by divineaverage** [comment]
Yeah, backend is running still (restarted just to be sure). I am just at a loss, I thought I updated the user slice right. But maybe I'm seeing this all wrong - I'm updating the user slice, should I be updating the post slice? Redux confuses me a little in this regard lol

**Comment by AccessiT3ch** [comment]
@divineaverage [this might be an issue](https://github.com/divineaverage/P7_Groupomania/blob/main/client/src/components/feed/Card.jsx#L20-L25):

```js
  // getProfileById is being used twice here
  const profileState = getProfileById(store.getState().profile, userState._id) || {}; // this is the profile object you want
  const [profile, setProfile] = useState(getProfileById(profileState, authorId)); // this is being called with something that's not the profile slice state
```

^ This should be simplified, something like:

```js  
const theProfile = getProfileById( store.getState().profile, userState._id );
const [profile, setProfile] = useState( theProfile || {} );

// or

const [profile, setProfile] = useState( getProfileById( store.getState().profile, userState._id ) || {} );

```

**Comment by divineaverage** [comment]
Definitely makes sense. Still doesn't work, unfortunately.
